amethyst_core/src/timing.rs

```diff
@@ -378,3 +378,8 @@ pub fn duration_to_nanos(duration: Duration) -> u64 {
 pub fn nanos_to_duration(nanos: u64) -> Duration {
     Duration::new(nanos / 1_000_000_000, (nanos % 1_000_000_000) as u32)
 }
+
+/// Multiplies a given Duration by the float provided.
+pub fn duration_f32_mul(duration: Duration, float: f32) -> Duration {
+    Duration::new((duration.as_secs() as f32 * float) as u64, (duration.subsec_nanos() as f32 * float) as u32)
```

Comment:
That's why I don't like `Duration`.
Can we consider to replace it with our own type?

Reply:
You can always use secs f64 if you don't want to use Duration.

Reply:
I'm leaving this function since I've already written it, but I'm not using it in this PR anymore.

Reply:
@jojolepro how many seconds I can store in `f64` until error become more than epsilon?
I'm strongly against using floats where error may become observable.

Reply:
Relax, an f32 has much higher precision than we need in this circumstance.  You only really need an f64 when storing very large values, which we definitely are not.  Most of these values are going to be below 1.0.